main.go

```diff
@@ -46,8 +45,42 @@ import (
 	"github.com/prometheus/pushgateway/storage"
 )
 
+var (
+	requestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "pushgateway_http_request_duration_seconds",
+			Help:    "Histogram of latencies for HTTP requests.",
+			Buckets: []float64{.05, 0.1, .25, .5, .75, 1, 2, 5, 20, 60},
+		},
+		[]string{"handler", "method"},
+	)
+	responseSize = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "pushgateway_http_response_size_bytes",
+			Help:    "Histogram of response size for HTTP requests.",
+			Buckets: prometheus.ExponentialBuckets(100, 10, 7),
+		},
+		[]string{"handler", "method"},
+	)
+)
+
 func init() {
 	prometheus.MustRegister(version.NewCollector("pushgateway"))
+	prometheus.MustRegister(requestDuration)
+	prometheus.MustRegister(responseSize)
+}
+
+func instrumentHandler(prefix string) func(handlerName string, handler http.HandlerFunc) http.HandlerFunc {
+	return func(handlerName string, handler http.HandlerFunc) http.HandlerFunc {
+		handlerLabel := prometheus.Labels{"handler": prefix + handlerName}
+		return promhttp.InstrumentHandlerDuration(
```

Comment:
As said above, I'd only do the instrumentation with `httpCnt` from `handler/metrics.go` for now. Please try to simplify the handler label (in line 75 above) so that we end up with the same label values as in the existing instrumentation (for the handlers that are already instrumented), i.e. `delete`, `wipe`, `status`, `healthy`, `ready`, `static`, `push` (and the new ones in the same style). With the code as is, you get fairly long and detailed labels that aren't necessarily helpful (e.g. `handler="/metrics/job/:job/*labels"` instead of just `handler="push"`). I know that the common/route package doesn't make that easy as it insists on using the path for `handlerName`, but at the very least we should not break current users by changing the existing handler label. Sorry for that…

Reply:
Quite confused. How I can implement this feature but for a temporary solution I am manual adding it.

Reply:
@beorn7 Any advice

Reply:
I think I have an idea. I'll show you in a separate PR. It's too hard to explain in the comments. And I would like to do it based on an older commit of yours.